static/app/views/settings/projectSecurityHeaders/projectCspReports.spec.tsx

```diff
@@ -0,0 +1,124 @@
+import {initializeOrg} from 'sentry-test/initializeOrg';
+import {render, screen, userEvent} from 'sentry-test/reactTestingLibrary';
+
+import ProjectCspReports from 'sentry/views/settings/projectSecurityHeaders/csp';
+
+describe('ProjectCspReports', function () {
+  const {project, organization, router} = initializeOrg();
+
+  const projectUrl = `/projects/${organization.slug}/${project.slug}/`;
+  const routeUrl = `/projects/${organization.slug}/${project.slug}/csp/`;
+
+  beforeEach(function () {
+    MockApiClient.clearMockResponses();
+    MockApiClient.addMockResponse({
+      url: `/projects/${organization.slug}/${project.slug}/keys/`,
+      method: 'GET',
+      body: [],
+    });
+    MockApiClient.addMockResponse({
+      url: projectUrl,
+      method: 'GET',
+      body: {
+        options: {},
+      },
+    });
+  });
+
+  it('renders', function () {
+    const {container} = render(
+      <ProjectCspReports
+        route={{}}
+        routeParams={router.params}
+        router={router}
+        routes={router.routes}
+        location={TestStubs.location({pathname: routeUrl})}
+        params={{orgId: organization.slug, projectId: project.slug}}
+      />
+    );
+    expect(container).toSnapshot();
+  });
+
+  it('can enable default ignored sources', function () {
+    render(
+      <ProjectCspReports
+        route={{}}
+        routeParams={router.params}
+        router={router}
+        routes={router.routes}
+        location={TestStubs.location({pathname: routeUrl})}
+        params={{orgId: organization.slug, projectId: project.slug}}
+      />
+    );
+
+    const mock = MockApiClient.addMockResponse({
+      url: projectUrl,
+      method: 'PUT',
+    });
+
+    expect(mock).not.toHaveBeenCalled();
+
+    // Click Regenerate Token
+    userEvent.click(
+      screen.getByRole('checkbox', {
+        name: 'Use default ignored sources Our default list will attempt to ignore common issues and reduce noise.',
+      })
+    );
+
+    expect(mock).toHaveBeenCalledWith(
+      projectUrl,
+      expect.objectContaining({
+        method: 'PUT',
+        data: {
+          options: {
+            'sentry:csp_ignored_sources_defaults': true,
+          },
+        },
+      })
+    );
+  });
+
+  it('can set additional ignored sources', function () {
+    render(
+      <ProjectCspReports
+        route={{}}
+        routeParams={router.params}
+        router={router}
+        routes={router.routes}
+        location={TestStubs.location({pathname: routeUrl})}
+        params={{orgId: organization.slug, projectId: project.slug}}
+      />
+    );
+
+    const mock = MockApiClient.addMockResponse({
+      url: projectUrl,
+      method: 'PUT',
+    });
+
+    expect(mock).not.toHaveBeenCalled();
+
+    // Click Regenerate Token
```

Comment:
Looks like this is typing not clicking

Reply:
oh yes.. good catch... I might have only copied/paste

Reply:
no no...it was there before and I think it makes sense to delete it

Reply:
whoops lol